src/mlpack/methods/ann/layer/multihead_attention_impl.hpp

```diff
@@ -132,28 +132,41 @@ Forward(const MatType& input, MatType& output)
 
   // Calculate the scores i.e. perform the matrix multiplication operation
   // on qProj and kProj. Here score = qProj . kProj'
-  scores = MultiplyCube2Cube(qProj, kProj, false, true);
+  scores = MultiplyCube2Cube(kProj, qProj, false, true);
 
   // Apply the attention mask if provided. The attention mask is used to black-
   // out future sequences and generally used in Encoder-Decoder attention.
   // The attention mask has elements -inf or 0.
   // The shape of the attention mask : (tgtSeqLen, srcSeqLen).
   if (!attnMask.is_empty())
   {
-    if (attnMask.n_rows != tgtSeqLen || attnMask.n_cols != srcSeqLen)
+    if (attnMask.n_slices != input.n_cols || attnMask.n_rows != tgtSeqLen || attnMask.n_cols != srcSeqLen)
       Log::Fatal << "The size of the 'attn_mask' is not correct.\n";
-    scores.each_slice() += attnMask;
+    for (size_t i = 0; i < batchSize; ++i)
+    {
+      for (size_t h = 0; h < numHeads; ++h)
+      {
+          scores.slice(i * numHeads + h) += attnMask.slice(i).t();
```

Comment:
If we're always transposing the attention mask, can we just change its definition to be size `(srcSeqLen, tgtSeqLen, batchSize)` instead?

(Also this line should only be two-space indented, not four.)